cpp/include/cudf/detail/join.hpp

```diff
@@ -18,6 +18,7 @@
 #include <cudf/column/column.hpp>
 #include <cudf/detail/structs/utilities.hpp>
 #include <cudf/detail/utilities/hash_functions.cuh>
+#include <cudf/table/experimental/row_operators.cuh>
```

Comment:
Recommend to use forward declaration of `preprocessed_table` so we don't have to include this header.

Reply:
```suggestion
namespace cudf::experimental::row::equality {
  class preprocessed_table;
}

```